### PR TITLE
Add Docker section to the Installation guide

### PR DESCRIPTION
**Pull request by antalszava**

### CODE REVIEW

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -49,15 +49,34 @@
   For example, if we wanted to only push single-qubit gates through
   controlled gates and cancel adjacent inverses, we could do:
 
+  ```python
+  from pennylane.transforms import commute_controlled, cancel_inverses
+  pipeline = [commute_controlled, cancel_inverses]
+
+  @qml.qnode(dev)
+  @qml.compile(pipeline=pipeline)
+  def qfunc(x, y, z):
```
Comment (antalszava) [inline .github/CHANGELOG.md:58]: The previous `qfunc` definition was decorated with `@qml.compile`, hence we need to redefine `qfunc`.

File: .github/CHANGELOG.md [hunk 2]
```diff
@@ -49,15 +49,34 @@
...
-  >>> compiled_qfunc = qml.compile(pipeline=pipeline)(qfunc)
-  >>> compiled_qnode = qml.QNode(compiled_qfunc, dev)
-  >>> print(qml.draw(compiled_qnode)(0.2, 0.3, 0.4))
-   0: ──H───RX(0.4)──RX(0.2)─────────────────────────────┤ ⟨Z⟩
-   1: ──H──╭X────────────────────────────────────╭CY─────┤
-   2: ──H──╰C────────RZ(0.4)──RZ(-0.4)──RX(0.3)──╰CY──Y──┤
+  >>> print(qml.draw(qfunc)(0.2, 0.3, 0.4))
+   0: ──H───RX(0.4)──RX(0.2)────────────────────────────┤ ⟨Z⟩
+   1: ──H──╭X───────────────────────────────────────╭C──┤
+   2: ──H──╰C────────RZ(0.4)──RZ(-0.4)──RX(0.3)──Y──╰Z──┤
```
Comment (antalszava) [inline .github/CHANGELOG.md:79]: We adjust the output to `CZ` instead of `CY` at the end.

**Comment by antalszava** [comment]
Docker changes in the installation guide of the Developer docs: https://pennylane--1538.org.readthedocs.build/en/1538/development/guide/installation.html#docker

File: doc/development/guide/installation.rst [hunk 3]
```diff
@@ -117,3 +117,77 @@ importing PennyLane in Python.
...
+
+Build a PennyLane Docker image
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+**Docker** support exists for building using **CPU** and **GPU** (Nvidia CUDA 11.1+) images.
+
+.. note::
+
+    Docker builds using "make" will work on Linux and MacOS only. For MS Windows
+    you can use [WSL](https://docs.microsoft.com/en-us/windows/wsl/install-win10).
```
Comment (josh146) [inline doc/development/guide/installation.rst:132]:
```suggestion
    you can use `WSL <https://docs.microsoft.com/en-us/windows/wsl/install-win10>`__.
```

**Review by josh146 — APPROVED** [review]

**Pull request merged**